### DISCUSSION TITLE
[New Feature] EMEffect is not cumulative

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1514] [Feature Requests]
### Description:
If a unit disabled by an EM pulse is struck by another EMP, the latter's stun duration overrides the former. It would make more sense for the EMP effect to stack, or at least the longer duration should override the shorter one. This would also prevent players from cheesing their way out of a stunlock with a shorter stun.

For reference, cumulative EMP is implemented in [Ares](http://ares-developers.github.io/Ares-docs/restored/emp.html).

### Conditions:
In vanilla, both sources of EMP (EMP Cannon and Mobile EMP) have the same stun duration, so this issue is not apparent. It is necessary to modify one of them to reproduce the bug:
```
[MobileEMPulseWeapon]
Damage=1
```

### Steps To Reproduce:
1. Modify the Mobile EMP's stun duration as noted above.
2. Hit two units with an EMP Cannon simultaneously.
3. Hit one of the units with Mobile EMP while it's still stunned by the EMP Cannon.
4. Watch as the unit hit by the Mobile EMP will recover much faster.

**Comment by Rampastring** [comment]
I think this is more of a feature request than a bug fix, as the question whether the EMP effect should stack or be overridden is a question of game design and balance. I think a new key for controlling the behaviour should be added.

**Comment by CCHyper** [comment]
Agreed, changing this to a feature request.
